src/get_timezone.py

```diff
@@ -37,30 +52,23 @@ async def get_timezone_from_location(update: Update, context: CallbackContext):
     user_timezone = TimezoneFinder().timezone_at(
         lng=update.message.location.longitude, lat=update.message.location.latitude
     )
-    if user_timezone is not None:
-        time_zone = pytz.timezone(user_timezone)
-        utc_time = datetime.datetime.utcnow()
-        utc = float(time_zone.utcoffset(utc_time).total_seconds() / 3600)
-        hours, minutes = divmod(utc * 60, 60)
-        utc = f"{hours:+03.0f}:{minutes:02.0f}"
-        text_utc = TIME_ZONE + utc
-        await set_timezone(update.effective_chat.id, text_utc, context)
-        return text_utc
-    return None
+    if user_timezone is None:
+        return None
+    time_zone = pytz.timezone(user_timezone).localize(datetime.datetime.now()).strftime("%z")
+    text_utc = TIME_ZONE + time_zone[:3] + time_zone[3:]
+    await set_timezone(update.effective_chat.id, text_utc, context)
+    return text_utc
 
 
 async def get_timezone_from_text_message(update: Update, context: CallbackContext):
     """
     Sets timezone based on a text message from the user.
     Return None if error, any else (string with timezone will be best).
     """
-    timezone = re.search(r"(UTC)?([-+]?)(\d{1,2})(:\d{1,2}|)$", update.message.text, flags=re.IGNORECASE)
+    timezone = get_timezone_utc_format(update.message.text)
     if timezone is None:
```

Comment:
```suggestion
    if timezone:
```
В других местах тоже можно поправить, вроще как леге воспринимать

Reply:
Тут, как бы, вопрос:
если бы была строка `if timezone is not None`, то заменить её на `if timezone` - святое дело...
Здесь ты предлагаешь по факту поменять блоки if-else местами, но тогда будет _больше_ строк со сдвигом. 
Я имею ввиду, что после `if timezone is None:` сейчас следует, как правило, одна строка с return, а далее основной код продолжается без отступа.
Я могу поменять, но, мне кажется, "легче воспринимать" - не про данную ситуацию.

Reply:
Поддерживаю. Читаемость выше, если мы обрубаем выполнение функции как у Олега сделано.

Reply:
Ой, извиняюсь, это я затупил